src/base/search/searchpluginmanager.cpp

```diff
@@ -322,13 +346,16 @@ QString SearchPluginManager::pluginsLocation()
 
 QString SearchPluginManager::engineLocation()
 {
-    QString folder = "nova";
-    if (Utils::ForeignApps::pythonInfo().version.majorNumber() >= 3)
-        folder = "nova3";
-    const QString location = Utils::Fs::expandPathAbs(specialFolderLocation(SpecialFolder::Data) + folder);
-    QDir locationDir(location);
-    if (!locationDir.exists())
+    static QString location;
```

Comment:
It's better to add SearchPluginManager::m_location instead.

Reply:
I don't mind the change, however a bunch of static functions rely on this one, i.e. static functions cannot access the class private variable. So just keep this way for now?

Reply:
I meant use static class variable instead of this static variable. What's the problem?
>static functions cannot access the class private variable

Why? The class static functions can access private members of the same class.

Reply:
>I meant use static class variable instead of this static variable. What's the problem?

I misunderstood you, you should have say so earlier.

Now I think of it, I don't know why it is better to have a static variable in class rather in function?

```c++
QString SearchPluginManager::engineLocation()
{
    if (m_engineLocation.isEmpty()) {
        const QString folder = (Utils::ForeignApps::pythonInfo().version.majorNumber() >= 3)
            ? "nova3" : "nova";
        m_engineLocation = Utils::Fs::expandPathAbs(specialFolderLocation(SpecialFolder::Data) + folder);

        const QDir locationDir(m_engineLocation);
        locationDir.mkpath(locationDir.absolutePath());
    }

    return m_engineLocation;
}
```

Reply:
Okay, whatever... It solves nothing at this time.